lib/request.coffee

```diff
@@ -45,12 +46,26 @@ prepareOptions = (options = {}) ->
 		gzip: true
 		json: true
 		headers: {}
+		refreshToken: true
 
 	options.url = url.resolve(settings.get('remoteUrl'), options.url)
 
-	utils.getAuthorizationHeader().then (authorizationHeader) ->
-		options.headers.Authorization = authorizationHeader
-		return options
+	Promise.try ->
+		return if not options.refreshToken
+
+		utils.shouldUpdateToken().then (shouldUpdateToken) ->
+			return if not shouldUpdateToken
+
+			exports.send
+				url: '/whoami'
+				refreshToken: false
+			.get('body').then (newToken) ->
+				token.set(newToken)
+
+	.then ->
```

Comment:
I personally prefer
```
.then(utils.getAuthorizationHeader)
.then (authorizationHeader) ->
```
or

```
.then ->
	utils.getAuthorizationHeader()
.then (authorizationHeader) ->
```
as it avoids the need for the additional indentation